proparse/src/main/java/org/prorefactor/proparse/antlr4/PostLexer.java

```diff
@@ -0,0 +1,305 @@
+    int currLine = currToken.getLine();
+    int currCol = currToken.getCharPositionInLine();
+    PreproIfState preproIfState = new PreproIfState();
+    preproIfVec.add(preproIfState);
+    // Only evaluate if we aren't consuming from an outer &IF.
+    boolean isTrue = preproIfCond(!prepro.isConsuming());
+    if (isTrue) {
+      prepro.getLstListener().preproIf(currLine, currCol, true);
+      preproIfState.done = true;
+    } else {
+      prepro.getLstListener().preproIf(currLine, currCol, false);
+      preproIfState.consuming = true;
+      preproconsume();
+    }
+  }
+
+  private void preproElse() throws IOException, TokenStreamException, RecognitionException {
+    LOGGER.trace("Entering preproElse()");
+
+    PreproIfState preproIfState = preproIfVec.getLast();
+    if (!preproIfState.done) {
+      preproIfState.consuming = false;
+      prepro.getLstListener().preproElse(currToken.getLine(), currToken.getCharPositionInLine());
+    } else {
+      if (!preproIfState.consuming) {
+        prepro.getLstListener().preproElse(currToken.getLine(), currToken.getCharPositionInLine());
+        preproIfState.consuming = true;
+        preproconsume();
+      }
+      // else: already consuming. no change.
+      prepro.getLstListener().preproElse(currToken.getLine(), currToken.getCharPositionInLine());
+    }
+  }
+
+  private void preproElseif() throws IOException, TokenStreamException, RecognitionException {
+    LOGGER.trace("Entering preproElseif()");
+    // Preserve the current position for listing, before evaluating the expression.
+    // We can't just write to listing here, because the expression evaluation may
+    // find macro references to list.
+    int currLine = currToken.getLine();
+    int currCol = currToken.getCharPositionInLine();
+    boolean evaluate = true;
+    // Don't evaluate if we're consuming from an outer &IF
+    if (prepro.getConsuming() - 1 > 0)
+      evaluate = false;
+    // Don't evaluate if we're already done with this &IF
+    if (preproIfVec.getLast().done)
+      evaluate = false;
+    boolean isTrue = preproIfCond(evaluate);
+    prepro.getLstListener().preproElseIf(currLine, currCol);
+    PreproIfState preproIfState = preproIfVec.getLast();
+    if (isTrue && (!preproIfState.done)) {
+      preproIfState.done = true;
+      preproIfState.consuming = false;
+    } else {
+      if (!preproIfState.consuming) {
+        preproIfState.consuming = true;
+        preproconsume();
+      }
+      // else: already consuming. no change.
+    }
+  }
+
+  private void preproEndif() throws IOException {
+    LOGGER.trace("Entering preproEndif()");
+    prepro.getLstListener().preproEndIf(currToken.getLine(), currToken.getCharPositionInLine());
+    // XXX Got a case where removeLast() fails with NoSuchElementException
+    if (!preproIfVec.isEmpty())
+      preproIfVec.removeLast();
+  }
+
+  private boolean preproIfCond(boolean evaluate) throws IOException, TokenStreamException, RecognitionException {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "preproIfCond" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)